src/vs/platform/theme/common/colorRegistry.ts

```diff
@@ -88,9 +88,10 @@ export interface IColorRegistry {
 	 * Register a color to the registry.
 	 * @param id The color id as used in theme description files
 	 * @param defaults The default values
+	 * @param needsTransparency Whether the color requires transparency
 	 * @description the description
 	 */
-	registerColor(id: string, defaults: ColorDefaults, description: string): ColorIdentifier;
+	registerColor(id: string, defaults: ColorDefaults, description: string, needsTransparency?: boolean): ColorIdentifier;
```

Comment:
I wouldn't expect an option boolean parameter to `registerColor` to control linting settings, so I'd make it an option:
```suggestion
	registerColor(id: string, defaults: ColorDefaults, description: string, options?: { needsTransparency?: boolean }): ColorIdentifier;
```

Reply:
@aeschli shall I do this consider we're exposing it on the registry interface now?